polar/src/polar.rs

```diff
@@ -248,7 +252,11 @@ impl Polar {
         for (sym, singleton) in singletons {
             if let Some(term) = singleton {
                 let mut writer = self.output.write().unwrap();
-                let _ = writeln!(&mut writer, "Singleton variable {}", sym);
+                let _ = if let Value::Pattern(..) = term.value() {
+                    writeln!(&mut writer, "Unknown specializer {}", sym)
+                } else {
+                    writeln!(&mut writer, "Singleton variable {}", sym)
```

Comment:
```suggestion
                    writeln!(&mut writer, "Singleton variable {}. This variable may be unused or undefined, see https://docs-preview.oso.dev/using/polar-syntax.html#variables ", sym)
```
We would need to update the link when the docs are live

Reply:
I strongly object to a warning that long. We could leave out the "Singleton variable" part, and make it `The variable {} is unused or undefined, see <...>`, but that's still pretty long. I really think it's best as-is, with an FAQ entry that points to the doc.

Reply:
OK, settled on a variant, we'll use a link shortener later.